src/cli/subcommands/car_cmd.rs

```diff
@@ -0,0 +1,125 @@
+// Copyright 2019-2023 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use std::path::PathBuf;
+
+use async_recursion::async_recursion;
+use clap::Subcommand;
+use futures::{AsyncRead, StreamExt};
+use fvm_ipld_car::Block;
+use fvm_ipld_car::CarHeader;
+use fvm_ipld_car::CarReader;
+use tokio_util::compat::TokioAsyncReadCompatExt;
+
+use crate::ipld::CidHashSet;
+
+#[derive(Debug, Subcommand)]
+pub enum CarCommands {
+    Concat {
+        /// A list of `.car` file paths
+        car_files: Vec<PathBuf>,
+        /// The output `.car` file path
+        #[arg(short, long)]
+        output: PathBuf,
+    },
+}
+
+impl CarCommands {
+    pub async fn run(self) -> anyhow::Result<()> {
+        match self {
+            Self::Concat { car_files, output } => {
+                let mut readers = Vec::with_capacity(car_files.len());
+                for f in car_files {
+                    readers.push(
+                        CarReader::new(
+                            tokio::io::BufReader::new(tokio::fs::File::open(f).await?).compat(),
+                        )
+                        .await?,
+                    );
+                }
```

Comment:
Streaming version just for comparison. I'm fine with using a for-loop, though.

```suggestion
                let readers: Vec<_> = stream::iter(car_files)
                    .then(|f| async {
                        CarReader::new(BufReader::new(File::open(f).await?).compat()).await
                    })
                    .try_collect()
                    .await?;
```